src/app/qgisapp.cpp

```diff
@@ -17581,3 +17602,47 @@ void QgisApp::showEvent( QShowEvent *event )
     }
   } );
 }
+
+void QgisApp::handleRenderedLayerStatistics( const QgsMapCanvas *mapCanvas ) const
+{
+  const QgsRenderedItemResults *renderedItemResults = mMapCanvas->renderedItemResults( false );
+  if ( !renderedItemResults )
+  {
+    return;
+  }
+
+  QList<const QgsRenderedLayerStatistics *> computedStatistics;
+  for ( const QgsRenderedItemDetails *item : renderedItemResults->renderedItems() )
+  {
+    if ( const QgsRenderedLayerStatistics *layerStatistics = dynamic_cast< const QgsRenderedLayerStatistics *>( item ) )
+    {
+      computedStatistics.append( layerStatistics );
+    }
+  }
+
+  if ( !computedStatistics.empty() )
+  {
+    for ( QgsMapLayer *layer : mMapCanvas->layers() )
+    {
+      if ( QgsRasterLayer *rasterLayer = qobject_cast<QgsRasterLayer *>( layer ) )
+      {
+        for ( const QgsRenderedLayerStatistics *statistics : computedStatistics )
+        {
+          if ( statistics->layerId() == rasterLayer->id() )
+          {
+
+            // refresh the renderer of the layer
+            rasterLayer->renderer()->refresh( statistics->boundingBox(), statistics->minimum(), statistics->maximum() );
```

Comment:
Shouldn't this be inside the if ( mapCanvas == mMapCanvas ) check? otherwise we'll end up changing the renderer for non-main canvas too!

Actually on that note, I don't think we should be calling handleRenderedLayerStatistics at ALL for non-main canvases.